source/extensions/filters/listener/http_inspector/http_inspector.cc

```diff
@@ -81,10 +88,14 @@ ParseState Filter::parseHttpHeader(absl::string_view data) {
 
     absl::string_view new_data = data.substr(nread_);
     const size_t pos = new_data.find_first_of("\r\n");
-
     if (pos != absl::string_view::npos) {
       // Include \r or \n
-      new_data = new_data.substr(0, pos + 1);
+      if (new_data[pos] == '\r' && (pos + 1 < new_data.size())) {
+        // Including '\n' as BalsaParser executes on seeing '\n'.
+        new_data = new_data.substr(0, pos + 2);
+      } else {
+        new_data = new_data.substr(0, pos + 1);
+      }
```

Comment:
should we to search the `\r\n` directly rather than to seach on of them?

Reply:
yes we can update the overall logic to search for `\r\n`, it will simply the check.

Reply:
> should we to search the `\r\n` directly rather than to seach on of them?

updated to search only for the `\n`, searching directly for `\r\n` is creats issues when both are not together.